src/java.base/share/classes/java/net/URI.java

```diff
@@ -544,6 +549,13 @@ public final class URI
     // The remaining fields may be computed on demand, which is safe even in
     // the face of multiple threads racing to initialize them
     private transient String schemeSpecificPart;
+
+    private static final ToIntFunction<URI> HASH_UPDATER = StableFieldUpdater.ofInt(
+            URI.class, "hash", new ToIntFunction<>() {
+                @ForceInline @Override public int applyAsInt(URI uri) { return uri.hashCode0(); }}, -1);
+
+    // Used reflectively by HASH_UPDATER
+    @Stable
```

Comment:
What are the effect of this change on startup time? I remember that @cl4es made a number of changes to URI to improve startup. Maybe this change should be dropped out of this PR for now.

Reply:
Yes. I think we should remove these public changes and maybe come back in a separate PR with those.